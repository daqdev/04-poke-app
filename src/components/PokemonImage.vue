<template>
  <img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon">
  <img class="fade-in" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon">
</template>

<script>
export default {

}
</script>

<style scoped>
.hidden-pokemon {
    filter: brightness(0);
}

</style>