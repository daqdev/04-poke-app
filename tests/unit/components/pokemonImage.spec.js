import { shallowMount } from "@vue/test-utils"
import PokemonImage from '@/components/PokemonImage'

describe('pokemonImage', ()=>{

    test('debe coincidir con el snapshot', ()=>{
        
        const wrapper = shallowMount(PokemonImage, {
            props:{
                pokemonId:199,
                showPokemon:false
            }

        })
        expect(wrapper.html()).toMatchSnapshot()


        // shallowMount.
        // shallowMount
        // PokemonImage.describe
    })

    test('debe mostrar la imagen oculta y el Pokemon nro 100', ()=>{

        const wrapper = shallowMount(PokemonImage, {
            props:{
                pokemonId:100,
                showPokemon:false
            }
        })
        
        const img1 = wrapper.find('img')
        expect(img1.classes('hidden-pokemon')).toBeTruthy()
        expect(img1.attributes('src').includes('100')).toBeTruthy()
    })    
    
    test('debe mostrar el pokemon si showPokemon es true', ()=>{

        const wrapper = shallowMount(PokemonImage, {
            props:{
                pokemonId:100,
                showPokemon:true
            }
        })
        const img1 = wrapper.find('img')
        expect(img1.classes('hidden-pokemon')).toBeFalsy()
        expect(img1.classes('fade-in')).toBeTruthy()


    })
})