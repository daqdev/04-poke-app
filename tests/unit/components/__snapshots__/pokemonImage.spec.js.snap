// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemonImage debe coincidir con el snapshot 1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/199.svg" alt="pokemon"></div>`;
